shell/platform/windows/flutter_windows_view.cc

```diff
@@ -81,6 +82,26 @@ void UpdateVsync(const FlutterWindowsEngine& engine,
   }
 }
 
+/// Destroys a rendering surface that backs a Flutter view.
+void DestroyWindowSurface(const FlutterWindowsEngine& engine,
+                          egl::WindowSurface& surface) {
+  // EGL surfaces are used on the raster thread if the engine is running.
+  // There may be pending raster tasks that use this surface. Destroy the
+  // surface on the raster thread to avoid concurrent uses.
+  if (engine.running()) {
+    fml::AutoResetWaitableEvent latch;
+    engine.PostRasterThreadTask([&]() {
+      surface.Destroy();
+      latch.Signal();
```

Comment:
What does this `latch` accomplish? Does this prevent the engine from destructing before this task runs?

Reply:
Good question. This runs in the `FlutterWindowsView` destructor, which owns the `std::unique<egl::WindowSurface>`. The latch forced the destructor to wait until the surface was destroyed.

A cleaner version would be to just move the surface's ownership to the raster thread. I've updated it to that insead!